rust/sbp/src/json/ser.rs

```diff
@@ -74,6 +74,22 @@ where
     Ok(())
 }
 
+/// Serialize the given message as a JSON byte vector.
+pub fn to_value<M>(msg: &M) -> Result<serde_json::Value, JsonError>
+where
+    M: SbpMessage + Serialize,
+{
+    let mut frame = BytesMut::with_capacity(BUFLEN);
+    let mut payload = String::with_capacity(BUFLEN);
+
+    let output = JsonOutput {
+        common: get_common_fields(&mut payload, &mut frame, msg)?,
+        msg,
+    };
+
+    serde_json::to_value(output).map_err(|e| e.into())
+}
+
```

Comment:
Another shed color could be a function available in the root of the crate `sbp::to_json`. Either way agree we should have some kind of callout explaining how it's different than the `Serialize` impl on the message type

Reply:
Also I guess the return type would be more accurate as `serde_json::Map<String, serde_json::Value>` because they are always objects but maybe that's less convenient overall 🤷